eclair-core/src/main/scala/fr/acinq/eclair/DBChecker.scala

```diff
@@ -16,31 +16,35 @@
 
 package fr.acinq.eclair
 
+import fr.acinq.eclair.channel.Commitments
 import grizzled.slf4j.Logging
 
 import scala.util.{Failure, Success, Try}
 
-object DBCompatChecker extends Logging {
+object DBChecker extends Logging {
 
   /**
-    * Tests if the channels data in the DB are compatible with the current version of eclair; throws an exception if incompatible.
-    *
-    * @param nodeParams
-    */
-  def checkDBCompatibility(nodeParams: NodeParams): Unit =
+   * Tests if the channels data in the DB are compatible with the current version of eclair; throws an exception if incompatible.
+   *
+   * @param nodeParams
+   */
+  def checkChannelsDB(nodeParams: NodeParams): Unit = {
+    logger.info("listLocalChannels()")
     Try(nodeParams.db.channels.listLocalChannels()) match {
-      case Success(_) => {}
+      case Success(channels) =>
+        channels.foreach(hasCommitments => if (!Commitments.validateSeed(hasCommitments.commitments, nodeParams.channelKeyManager)) throw InvalidChannelSeedException(hasCommitments.channelId))
```

Comment:
nit: we've gotten rid of the `HasCommitments` trait:
```suggestion
        channels.foreach(data => if (!Commitments.validateSeed(data.commitments, nodeParams.channelKeyManager)) throw InvalidChannelSeedException(data.channelId))
```